pyrokinetics/equilibrium/equilibrium.py

```diff
@@ -363,15 +371,26 @@ def __init__(
         # Create normalised 1d psi grid
         psi_n = (psi - psi[0]) / (psi_lcfs - psi[0])
 
+        index_direction = Ip_sign
         # Create spline functions for all psi grids with respect to psi
-        self._F_psi_spline = UnitSpline(psi, F)
-        self._FF_prime_psi_spline = UnitSpline(psi, FF_prime)
-        self._p_psi_spline = UnitSpline(psi, p)
-        self._p_prime_psi_spline = UnitSpline(psi, p_prime)
-        self._q_psi_spline = UnitSpline(psi, q)
-        self._R_major_psi_spline = UnitSpline(psi, R_major)
-        self._r_minor_psi_spline = UnitSpline(psi, r_minor)
-        self._Z_mid_psi_spline = UnitSpline(psi, Z_mid)
+        self._F_psi_spline = UnitSpline(psi[::index_direction], F[::index_direction])
+        self._FF_prime_psi_spline = UnitSpline(
+            psi[::index_direction], FF_prime[::index_direction]
+        )
```

Comment:
We could implement this behaviour in `UnitSpline` to keep this cleaner and avoid having to use `index_direction`. Currently the `__init__` is this:
```python
    def __init__(self, x: ArrayLike, y: ArrayLike):
        self._x_units = x.units
        self._y_units = y.units
        self._spline = InterpolatedUnivariateSpline(x.magnitude, y.magnitude)
```
It could be altered to this:
```python
    def __init__(self, x: ArrayLike, y: ArrayLike):
        self._x_units = x.units
        self._y_units = y.units
        x_mag = x.magnitude
        y_mag = y.magnitude
        # Assume x is monotonically increasing/decreasing
        if x_mag[1] > x_mag[0]:
            self._spline = InterpolatedUnivariateSpline(x_mag, y_mag)
        else:
            self._spline = InterpolatedUnivariateSpline(x_mag[::-1], y_mag[::-1])
```

Reply:
Yeah I thought this would work, wasn't sure if we wanted to be explicit in when we reverse, but this is probably fine